File: {{cookiecutter.project_slug}}/src/{{cookiecutter.project_slug}}/resources.py
{% if cookiecutter.sdk_choice == 'python' and cookiecutter.include_resources == 'yes' -%}
"""Resource implementations for {{ cookiecutter.project_name }}."""

import logging
from typing import List
from mcp.server import Server
from mcp.types import Resource, TextContent

logger = logging.getLogger(__name__)


def register_resources(server: Server):
    """Register all resources with the MCP server."""

    @server.list_resources()
    async def list_resources() -> List[Resource]:
        """List available resources."""
        return [
            Resource(
                uri="example://resource",
                name="Example Resource",
                description="Example resource - replace with your resources",
                mimeType="text/plain"
            )
        ]

    @server.read_resource()
    async def read_resource(uri: str) -> str:
        """Read a resource by URI."""
        logger.info(f"Reading resource: {uri}")

        if uri == "example://resource":
            return "Example resource content"

        raise ValueError(f"Unknown resource: {uri}")
{%- endif %}

File: {{cookiecutter.project_slug}}/src/resources.ts
{% if cookiecutter.sdk_choice == 'typescript' and cookiecutter.include_resources == 'yes' -%}
/**
 * Resource implementations for {{ cookiecutter.project_name }}
 */

import { Server } from "@modelcontextprotocol/sdk/server/index.js";
import { ListResourcesRequestSchema, ReadResourceRequestSchema } from "@modelcontextprotocol/sdk/types.js";

const log = {
  info: (msg: string) => console.error(`[INFO] ${msg}`),
};

export function registerResources(server: Server): void {
  // List available resources
  server.setRequestHandler(ListResourcesRequestSchema, async () => {
    return {
      resources: [
        {
          uri: "example://resource",
          name: "Example Resource",
          description: "Example resource - replace with your resources",
          mimeType: "text/plain",
        },
      ],
    };
  });

  // Read a resource
  server.setRequestHandler(ReadResourceRequestSchema, async (request) => {
    const { uri } = request.params;

    log.info(`Reading resource: ${uri}`);

    switch (uri) {
      case "example://resource":
        return {
          contents: [
            {
              uri,
              mimeType: "text/plain",
              text: "Example resource content",
            },
          ],
        };

      default:
        throw new Error(`Unknown resource: ${uri}`);
    }
  });
}
{%- endif %}
